GitHub API failed with status code 403

I have added the job from the README in a private repo to check conventional commits while creating a PR, but it failed with the following message: 

```
Run opensource-nepal/commitlint@v1
  with:
    fail_on_error: true
    verbose: false
    token: ***
Run actions/setup-python@v5.1.0
  with:
    python-version: 3.8
    check-latest: false
    token: ***
    update-environment: true
    allow-prereleases: false
Installed versions
  Version 3.8 was not found in the local cache
  Version 3.8 is available for downloading
  Download from "https://github.com/actions/python-versions/releases/download/3.8.18-12303122501/python-3.8.18-linux-24.04-x64.tar.gz"
  Extract downloaded archive
  /usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/b56b4199-7104-452f-b816-5a1903b188a1 -f /home/runner/work/_temp/4935eec0-c1a2-4666-8375-d583f2a681c0
  Execute installation script
  Check if Python hostedtoolcache folder exist...
  Create Python 3.8.18 folder
  Copy Python binaries to hostedtoolcache folder
  Create additional symlinks (Required for the UsePythonVersion Azure Pipelines task and the setup-python GitHub Action)
  Upgrading pip...
  Looking in links: /tmp/tmpfr01dzg1
  Requirement already satisfied: setuptools in /opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages (56.0.0)
  Requirement already satisfied: pip in /opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages (23.0.1)
  Collecting pip
  Downloading pip-25.0.1-py3-none-any.whl (1.8 MB)
  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/1.8 MB 33.8 MB/s eta 0:00:00
  Installing collected packages: pip
  Attempting uninstall: pip
  Found existing installation: pip 23.0.1
  Uninstalling pip-23.0.1:
  Successfully uninstalled pip-23.0.1
  Successfully installed pip-25.0.1
  Create complete file
  Successfully set up CPython (3.8.18)
Run python -m pip install --quiet --disable-pip-version-check -e ${GITHUB_ACTION_PATH}
  python -m pip install --quiet --disable-pip-version-check -e ${GITHUB_ACTION_PATH}
  python /home/runner/work/_actions/opensource-nepal/commitlint/v1/github_actions
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.8.18/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.8.18/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.8.18/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.8.18/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.8.18/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.8.18/x64/lib
    INPUT_TOKEN: ***
    INPUT_FAIL_ON_ERROR: true
    INPUT_VERBOSE: false
Error: Github API failed with status code 403
Error: Process completed with exit code 1.
```

--- a/README.md
+++ b/README.md
@@ -32,19 +32,41 @@ name: Conventional Commitlint
 
 on:
   push:
-    branches: ['main']
+    branches: ["main"]
   pull_request:
 
 jobs:
   commitlint:
     runs-on: ubuntu-latest
     name: Conventional Commitlint
+    permissions:
+      contents: read
+      pull-requests: read
     steps:
       - name: Conventional Commitlint
         uses: opensource-nepal/commitlint@v1
 ```
 
 > **_Note:_** The `commitlint` GitHub Action is triggered only by `push`, `pull_request`, or `pull_request_target` events.
+
+#### GitHub Action Permissions
+
+The action requires read permission for the following scopes:
+
+- `contents: read` : This allows the action to list commits of the repository.
+- `pull-requests: read` : This allows the action to list the commits in a Pull Request.
+
+For public repositories, these permissions are granted by default.
+For private repositories, you must explicitly grant them, either in:
+
+- the [repository’s default workflow permissions](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#configuring-the-default-github_token-permissions), or
+- the [workflow/job permissions block](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#permissions)
+
+Without these, you may encounter an error like:
+
+```
+Github API failed with status code 403. Response: {'message': 'Resource not accessible by integration', 'documentation_url': 'https://docs.github.com/rest/pulls/pulls#list-commits-on-a-pull-request', 'status': '403'}
+```
 
 #### GitHub Action Inputs
 

